src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -743,6 +747,70 @@ void GZBridge::navSatCallback(const gz::msgs::NavSat &nav_sat)
 	pthread_mutex_unlock(&_node_mutex);
 }
 
+bool GZBridge::callEntityFactoryService(const std::string &service, const gz::msgs::EntityFactory &req)
+{
+	bool result;
+	gz::msgs::Boolean rep;
+
+	if (_node.Request(service, req, 1000, rep, result)) {
+		if (!rep.data() || !result) {
+			PX4_ERR("EntityFactory service call failed.");
+			return PX4_ERROR;
```

Comment:
Does the PX4_ERROR evaluate to false? Perhaps it would make sense to return false, since the if statement evaluating the callEntityFactoryService already returns a PX4_ERROR if it's wrong.

Reply:
This is a mistake, should have been false